core/src/main/kotlin/com/samples/verifier/internal/SamplesVerifierInstance.kt

```diff
@@ -129,9 +154,37 @@ internal class SamplesVerifierInstance(compilerUrl: String, kotlinEnv: KotlinEnv
         } else emptyList()
       }
     }.withIndex().map { code ->
-      CodeSnippet("${file.nameWithoutExtension}_${code.index}", code.value)
+      CodeSnippet(file.toRelativeString(baseDir), code.value)
     }
   }
 
+  private fun processRepoFiles(repository: Repository, commit: RevCommit, filenames: List<String>, type: FileType): List<CodeSnippet> {
+    return filterFiles(filenames)
+      .flatMap {
+        val content = extractFiles(repository, commit, listOf(it))
```

Comment:
Why not call extractFiles with filenames once before map?

Reply:
Then content of all files will be kept in memory. Currently only one file content is in memory.